tensorflow_model_optimization/python/core/sparsity_tf2/lthpruner_test.py

```diff
@@ -349,50 +343,59 @@ def testReloadWeightsatInitialization(self):
         block_size=self.block_size,
         block_pooling_type=self.block_pooling_type)
 
-    optimizer = self.dummy_optimizer
+    optimizer =   tf.keras.optimizers.SGD(learning_rate=0.01)
     optimizer.iterations.assign(0)
     expected_saved_initialization = None
 
-    def _train(weight):
+    def _train(optimizer, weight):
       expected_saved_initialization = tf.ones_like(weight, dtype=weight.dtype.base_dtype) * -1
+      expected_reload = tf.ones_like(weight, dtype=weight.dtype.base_dtype) * -1
       p.create_slots(optimizer, weight)
       for i in tf.range(0, 2):
         p.preprocess_weights(optimizer, weight, self.grad(weight))
         if i == save_round:
           expected_saved_initialization = weight.read_value()
         weight.assign(tf.math.add(weight, sample_noise(i)))
         p.postprocess_weights(optimizer, weight, self.grad(weight))
+        should_reload = p._reload_schedule._should_prune_in_step(optimizer.iterations,
```

Comment:
As written this is checking implementation, not behavior. (It's checking a bunch of private attributes of the pruner and the pruning schedule)

can we make this test have a reload_step specified at the start w/ save_step, and we directly compare it to `i` like how we do for save round?